ovos_utils/gui.py

```diff
@@ -429,14 +429,19 @@ class GUIInterface:
     """
 
     def __init__(self, skill_id, bus=None, remote_server=None, config=None):
-        self.bus = bus or get_mycroft_bus()
+        self.bus = bus
         self.__session_data = {}  # synced to GUI for use by this skill's pages
         self.pages = []
         self.current_page_idx = -1
         self.skill_id = skill_id
         self.on_gui_changed_callback = None
         self.remote_url = remote_server
         self._events = []
+        if bus:
+            self.set_bus(bus)
```

Comment:
should this happen regardless? `if not bus... get_mycroft_bus()` should occur if `bus`-the-argument is optional and defaults to None

Reply:
usually the bus connection object is shared, this allows to set it only when needed. I had to postpone the init of the bus in OCP until i had constructed the player object and needed this

but the GUI can not work without a bus, it is always passed around.

Reply:
to be clear i needed to call `set_bus` after having constructed the gui object, in order to share the correct bus object from the player.

The aim is that if you pass bus=None a new connection is not automatically created, easy to forget to close those and leave them hanging forever.... so i want to reduce the automatic creation of bus connections as much as possible

Reply:
Sounds like what we really need is a flag on the bus connection that remembers its parent process, and commits suicide when the parent is missing.

But, in the meantime, fair enough.